pyproject.toml

```diff
@@ -75,16 +76,37 @@ icechunk = [
     "icechunk>=0.2.4",
 ]
 kerchunk = ["fastparquet"]
+
 all_writers = [
     "virtualizarr[icechunk]",
     "virtualizarr[kerchunk]",
 ]
-all = [
-    "virtualizarr[all_readers]",
-    "virtualizarr[all_writers]",
-]
 
-test = [
+# Dependency sets under dependencies-groups are NOT available via PyPI
+[dependency-groups]
+upstream = [
+    'xarray @ git+https://github.com/pydata/xarray',
+    'universal_pathlib @ git+https://github.com/fsspec/universal_pathlib',
+    'numcodecs @ git+https://github.com/zarr-developers/numcodecs',
+    'ujson @ git+https://github.com/ultrajson/ultrajson',
+    'zarr @ git+https://github.com/zarr-developers/zarr-python',
```

Comment:
Should zarr-python from github be an upstream dependency? I know it's also in default dependencies but just curious because zarr-python also a dependency of icechunk.

Reply:
Since we're using Zarr's internal metadata classes in the `develop` branch, I think it's important to explicitly test for compatibility with the latest, unreleased changes in Zarr-Python.